src/modules/x509/extensions.ts

```diff
@@ -0,0 +1,123 @@
+import {
+	AuthorityKeyIdentifierExtension,
+	BasicConstraintsExtension,
+	ExtendedKeyUsageExtension,
+	Extension,
+	KeyUsageFlags,
+	KeyUsagesExtension,
+	SubjectAlternativeNameExtension,
+	SubjectKeyIdentifierExtension,
+} from "@peculiar/x509";
+import { KEY_USAGE_NAMES } from "./constants";
+
+interface ExtensionConfig {
+	name: string;
+	toJSON: (rawExtension: Extension) => {};
+}
+
+const UTF_8_DECODER = new TextDecoder("utf-8");
+function textDecoder(rawExtension: Extension) {
+	return UTF_8_DECODER.decode(rawExtension.value);
+}
+
+/**
+ * Map from OID to Extension
+ */
+export const EXTENSIONS_CONFIG: Record<string, ExtensionConfig> = {
+	"2.5.29.14": {
+		name: "X509v3 Subject Key Identifier",
+		toJSON(rawExtension: Extension) {
+			const ext = new SubjectKeyIdentifierExtension(rawExtension.rawData);
+			return [
+				ext.keyId
+					.match(/.{1,2}/g)
+					?.join(":")
+					.toUpperCase(),
+			];
+		},
+	},
+	"2.5.29.15": {
+		name: "X509v3 Key Usage",
+		toJSON(rawExtension: Extension) {
+			const ext = new KeyUsagesExtension(rawExtension.rawData);
+			const usages: string[] = [];
+
+			const keys = Object.keys(KEY_USAGE_NAMES) as unknown as KeyUsageFlags[];
+			for (const key of keys) {
+				if (ext.usages & key) {
+					usages.push(KEY_USAGE_NAMES[key]);
+				}
+			}
+			return usages;
+		},
+	},
+	"2.5.29.17": {
+		name: "X509v3 Subject Alternative Name",
+		toJSON(rawExtension: Extension) {
+			return new SubjectAlternativeNameExtension(rawExtension.rawData).toJSON();
+		},
+	},
+	"2.5.29.19": {
+		name: "X509v3 Basic Constraints",
+		toJSON(rawExtension: Extension) {
+			const ext = new BasicConstraintsExtension(rawExtension.rawData);
+			return {
+				CA: ext.ca,
+			};
+		},
+	},
+	"2.5.29.35": {
+		name: "X509v3 Authority Key Identifier",
+		toJSON(rawExtension: Extension) {
+			const ext = new AuthorityKeyIdentifierExtension(rawExtension.rawData);
+			return {
+				keyid: ext.keyId
+					?.match(/.{1,2}/g)
+					?.join(":")
+					.toUpperCase(),
+				certid: ext.certId,
+			};
+		},
+	},
+	"2.5.29.37": {
+		name: "X509v3 Extended Key Usage",
+		toJSON(rawExtension: Extension) {
+			const ext = new ExtendedKeyUsageExtension(rawExtension.rawData);
+			return ext.usages.map(code => {
+				switch (code) {
+					case "1.3.6.1.5.5.7.3.3":
+						return "Code Signing";
+					default:
+						return code;
+				}
+			});
+		},
+	},
+	/**
+	 * Fulcio OIDs are based on https://github.com/sigstore/fulcio/tree/main/pkg/ca/x509ca/extensions.go
+	 */
+	"1.3.6.1.4.1.57264.1.1": {
+		name: "Fulcio Issuer",
```

Comment:
nit: Can we drop Fulcio from all of these names? Otherwise, it sounds like Fulcio decided the value, which is inaccurate, since the values come from a presented OIDC token.